### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ghost**
[DEMO LINK](https://KrisMur.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by AndriiNychaiuk — CHANGES_REQUESTED** [review]
Great job! Almost everything is working as it shoud! But there is few things you probably didn't expect. There are some todos without any user, so when I opened that todos endless loader appeared. And there are some users without any name. For better understanding of the issue I recorder a short video.
https://www.loom.com/share/1b7ae705b7054f2aa094a0ece93aaa7b

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,86 @@
...
+              <TodoFilter filteredTodos={filtredTodos} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoading
+                ? (
+                  <Loader />
+                ) : (
```
Comment (AndriiNychaiuk) [inline src/App.tsx:65]: It confuse a little. Why Loader renders when !isLoading?)

**Review by AndriiNychaiuk — APPROVED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,34 +1,86 @@
...
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo > 0 && (
+        <TodoModal
+          todo={visibleTodos.find(todo => todo.id === selectedTodo)}
+          selectUser={select}
```
Comment (Olexandr6) [inline src/App.tsx:79]: This callback for closing TodoModal
```suggestion
          onClose={select}
```

File: src/components/Loader/Loader.scss [hunk 3]
```diff
@@ -11,7 +11,7 @@
     margin: 1em auto;
     border: 0.3em solid #ddd;
     border-left-color: #000;
-    animation: load8 1.2s infinite linear;
+    animation: load8 4s infinite linear;
```
Comment (Olexandr6) [inline src/components/Loader/Loader.scss:14]: ?

Comment (Olexandr6) [reply src/components/Loader/Loader.scss:14]: I guess this change, produce this bug:
image

**Review by G-YaSha — CHANGES_REQUESTED** [review]
Like your solution👍 Let's improve some things, and please, pay attention to naming your functions and variables

File: src/App.tsx [hunk 4]
```diff
@@ -1,34 +1,86 @@
...
+    const todos = todosFromServer.filter(todo => {
+      switch (condition) {
+        case 'active':
+          return !todo.completed && todo.title.includes(query);
+
+        case 'completed':
+          return todo.completed && todo.title.includes(query);
+
+        default:
+          return todo.title.includes(query);
```
Comment (jstmpelowycz) [inline src/App.tsx:40]: you'd rather apply enum for cases

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**